oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+make organizing and using the up to date translations simpler.
+
+Current State
+*************
+
+* edx-transifex-bot is a potential security issue: The edx-transifex-bot requires admin
+  rights on Transifex in order to function. Admin rights give access to private/sensitive
+  information as well as the ability to permanently delete translation and configuration
+  files. At some point, the login to the edx-transifex-bot user was lost, and without
+  access to infrastructure that the bot uses to function, this edx-transifex-bot is a
+  security issue most of the Open edX community cannot control or debug.
+* edx-transifex-bot is a black box for most of the community: the code for the
+  edx-transifex-bot is in the `ecommerce-scripts`_ repository but it is impossible for
+  most of the community to observe the work it is doing, or whether it is doing it
+  correctly. In addition, there is no documentation for these important scripts.
+* The translations for the Open edX Maple Release were never uploaded to Transifex,
+  because the automation handled by the edx-transifex-bot never uploaded it.
+* The underlying transifex-client library and Transifex API (V2) are being deprecated on
+  November 30th, 2022. Prior to removing the transifex-client as a dependency, this led
+  to inconsistent behavior by our tooling when we try to automatically manage
+  translations. See `this pull request`_ for more details.
+* We have a complex process for managing translations for the named releases. As a
+  result, the black box nature of the edx-transifex-bot and the unclear state of the
+  underlying tooling, this has become more laborious to keep running. Especially because
+  there are few people with Admin rights to Transifex and knowledge of the Transifex API;
+  this could become a recurring problem with each Open edX release.
+
+.. _ecommerce-scripts: https://github.com/openedx/ecommerce-scripts/tree/master/transifex
+.. _this pull request: https://github.com/openedx/edx-platform/pull/30567
+
+Rationale for migrating to the `Transifex GitHub App`_
+******************************************************
+
+* This is an upgrade of a system we use regularly, but do not want to have to maintain
+  regularly.
+* Upgrading from a bot (machine user) to an app/workflow is recommended by GitHub and
+  makes the translation process more open source. 
+* The `Transifex GitHub App`_ is developed and supported by Transifex
+* The `Transifex GitHub App`_ is very simple to configure and has many options. We can
+  set Transifex Projects to automatically upload/download translation files from a
+  repository once the translations are reviewed and accepted.
+* By using an app that is maintained by Transifex the organization, we reduce the
+  maintenance burden and are more future proof of changes they might make since they
+  maintain both the API and the `Transifex GitHub App`_.
+
+Rationale for consolidating translations files centrally
```

Comment:
I'm curious:

(1) What are you asking for?
(2) What "powers" do you want and why?
(3) What "powers" do you currently exercise?

Reply:
The Translation WG is in charge of translating and reviewing translations, and to my knowledge these translations are just automatically merged in, e.g. https://github.com/openedx/paragon/pull/1744. With the changes proposed in OEP-58, Translations WG would still be in charge of translating and reviewing translations. Approval limits (once it hits X% translated it is automatically committed to openedx-translations) can be set per translation file. I can see a situation where maintainers request the Translation WG to set the approval limit below the standard set by the Translations WG. That being said, our Translators are top knotch and keep translations at 90% or higher.

I think answers to @sarina's questions can help shed light to what type of input maintainers are looking for.

Reply:
> What are you asking for?

I am trying to understand what the interaction should be between Open edX developers and/or maintainers with the Translation WG. For example, I currently do not know where to find information on the roles and responsibilities of maintainers, developers, and the Translation WG. It may be helpful to point to that information in this OEP.

> What "powers" do you want and why? What "powers" do you currently exercise?

> The Translation WG is in charge of translating and reviewing translations

This is new to me! I greatly appreciate the Translation WG's work, don't get me wrong. However, I thought the status quo was that maintainers ultimately owned their repo's translations. For example, I think many maintainers were routinely given Project Maintainer access to Transifex, and would frequently help add/change/remove strings or review translations. If the ownership of translations is changing or has already changed, it may be helpful to point to that change in this OEP.

Also, previously, maintainers were able to modify their translation files. Now, via this OEP, it appears those translation files are being moved to a centralized repository, and I am unsure who will own that repository and if any of the access maintainers previously had to change those files will change. In short, "will there ever be non-bot PRs for the translation files in openedx-translations?" and "who will approve and/or merge those PRs?".

Reply:
@pshiu brought up some important questions.

- Who will maintain the openedx-translations repository?
- How should errors/bugs in translations in the openedx-translations repository be fixed?
- Who is in charge of approving and/or merging PRs in the openedx-translations repository?

It may be unreasonable to expect any one group to maintain openedx-translations as there are lots of stakeholders. Translators are expected to translate, but may not get to see if their translations will cause errors. Developers may see that there are errors, but may not understand the language of the translation. There will have to be cooperation. 

We should set up an issue request form on openedx-translations (much like [the Github Requests in tcril-engineering](https://github.com/openedx/tcril-engineering/issues/new/choose)) that notifies the Translations WG on slack when there is a translation error that needs to be fixed. This way, Translators can be notified of an error in a way they will be able to digest it, and Developers can write the problem and proposed solution in a way that is common to them.

We will have to look further into how the GitHub Transifex app manages changes to translation files outside of Transifex (https://github.com/openedx/openedx-translations/issues/29) to confirm that making PRs is a possible fix that developers can make. In the case that it is allowed, we can treat it as any other repository and nominate Open edX Community members to be Core Contributors to openedx-translations based on their technical experience and participation in the Translations WG.

@brian-smith-tcril is currently looking at this issue and will update with the Transifex App's capability.

Reply:
[i commented on the issue on the openedx-translations repo](https://github.com/openedx/openedx-translations/issues/29#issuecomment-1319105569), but i'll reiterate here:

based on transifex docs https://help.transifex.com/en/articles/6265125-github-via-transifex-ui

> For existing resources that have already been synced with GitHub, if translations are found in GitHub, they will be sent to Transifex only if no translations are found for that language in Transifex

knowing this, it seems we should stick to the idea of transifex as a source of truth for all translated strings, and any updates should happen in transifex